openstf/model/objective_creator.py

```diff
@@ -34,11 +38,15 @@ def create_objective(model_type: Union[MLModelType, str]) -> RegressorObjective:
             # This will raise a ValueError when an invalid model_type str is used
             # and nothing when a MLModelType enum is used.
             model_type = MLModelType(model_type)
+            objective = ObjectiveCreator.OBJECTIVES[model_type]
         except ValueError as e:
-            valid_types = [t.value for t in MLModelType]
-            raise NotImplementedError(
-                f"No objective for '{model_type}', "
-                f"valid model_types are: {valid_types}"
-            ) from e
+            if is_custom_type(model_type):
```

Comment:
This custom type will become standard behavior. So it's nicer if we can load the custom objective in the try statement instead of in except statement. Then the except statement has the logic for error handling